service/controller/resource/tccp/create.go

```diff
@@ -237,52 +240,714 @@ func (r *Resource) getCloudFormationTags(cr v1alpha1.Cluster) []*cloudformation.
+					ResourceName: key.DockerVolumeResourceName(cr, t),
+				},
+			},
+			OperatorVersion: key.OperatorVersion(&cr),
+			Route53Enabled:  r.route53Enabled,
+		}
+	}
+
+	return outputs, nil
+}
+func newRecordSetsParams(ctx context.Context, cr v1alpha1.Cluster, t time.Time, r *Resource) (*template.ParamsMainRecordSets, error) {
+	_, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var recordSets *template.ParamsMainRecordSets
+	{
+		recordSets = &template.ParamsMainRecordSets{
+			BaseDomain:                 key.ClusterBaseDomain(cr),
+			EtcdDomain:                 key.ClusterEtcdEndpoint(cr),
+			ClusterID:                  key.ClusterID(&cr),
+			MasterInstanceResourceName: key.MasterInstanceResourceName(cr, t),
+			Route53Enabled:             r.route53Enabled,
+			VPCRegion:                  key.Region(cr),
+		}
+	}
+
+	return recordSets, nil
+}
+func newRouteTablesParams(ctx context.Context, cr v1alpha1.Cluster) (*template.ParamsMainRouteTables, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var publicRouteTableNames []template.RouteTableName
+	for _, az := range cc.Spec.TenantCluster.TCCP.AvailabilityZones {
+		rtName := template.RouteTableName{
+			AvailabilityZone:    az.Name,
+			ResourceName:        key.SanitizeCFResourceName(key.PublicRouteTableName(az.Name)),
+			VPCPeeringRouteName: key.SanitizeCFResourceName(key.VPCPeeringRouteName(az.Name)),
+		}
+		publicRouteTableNames = append(publicRouteTableNames, rtName)
+	}
+
+	var privateRouteTableNames []template.RouteTableName
+	for _, az := range cc.Spec.TenantCluster.TCCP.AvailabilityZones {
+		if az.Name != key.MasterAvailabilityZone(cr) {
+			continue
+		}
+
+		rtName := template.RouteTableName{
+			AvailabilityZone:    az.Name,
+			ResourceName:        key.SanitizeCFResourceName(key.PrivateRouteTableName(az.Name)),
+			VPCPeeringRouteName: key.SanitizeCFResourceName(key.VPCPeeringRouteName(az.Name)),
+		}
+		privateRouteTableNames = append(privateRouteTableNames, rtName)
+	}
+
+	var routeTables *template.ParamsMainRouteTables
+	{
+		routeTables = &template.ParamsMainRouteTables{
+			HostClusterCIDR:        cc.Status.ControlPlane.VPC.CIDR,
+			PrivateRouteTableNames: privateRouteTableNames,
+			PublicRouteTableNames:  publicRouteTableNames,
+		}
+	}
+
+	return routeTables, nil
+}
+
+func getMasterRules(cfg securityConfig, hostClusterCIDR string) ([]template.SecurityGroupRule, error) {
+	// Allow traffic to the Kubernetes API server depending on the API
+	// whitelisting rules.
+	publicAPIRules, err := getKubernetesPublicAPIRules(cfg, hostClusterCIDR)
+	if err != nil {
+		return []template.SecurityGroupRule{}, microerror.Mask(err)
+	}
+
+	// Other security group rules for the master.
+	otherRules := []template.SecurityGroupRule{
+		{
+			Description: "Allow traffic from control plane CIDR to 4194 for cadvisor scraping.",
```

Comment:
I know you just took it over as it was but I would strongly vote for getting this into the templates just as with the other resources. Fine with doing this later as well. Just spreading the word.